### PR TITLE
fix: handle incorrect ids when importing alert_configuration or project_ip_access_list resources

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Ticket: [INTMDB-1113](https://jira.mongodb.org/browse/INTMDB-1113)

- Handle incorrect ids when importing alert_configuration or project_ip_access_list resources. 
- Minor fix in error handling for project data source read.

## Type of change:

- [X] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [X] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [X] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [X] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [X] I have added any necessary documentation (if appropriate)
- [X] I have run make fmt and formatted my code

### CODE REVIEW

File: mongodbatlas/fw_resource_mongodbatlas_alert_configuration_test.go [hunk 1]
```diff
@@ -312,6 +313,33 @@ func TestAccConfigRSAlertConfiguration_importBasic(t *testing.T) {
...
+			{
+				Config: testAccMongoDBAtlasAlertConfigurationConfig(orgID, projectName, true),
+			},
+			{
+				ResourceName:            resourceName,
+				ImportStateIdFunc:       testAccCheckMongoDBAtlasAlertConfigurationImportStateIncorrectIDFunc(resourceName),
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"project_id"},
+				ExpectError:             regexp.MustCompile("import format error"),
```
Comment (gssbzn) [inline mongodbatlas/fw_resource_mongodbatlas_alert_configuration_test.go]: wouldn't this panic?

Comment (lantoli) [reply mongodbatlas/fw_resource_mongodbatlas_alert_configuration_test.go]: No, I tested it locally, why do you think so?

Comment (gssbzn) [reply mongodbatlas/fw_resource_mongodbatlas_alert_configuration_test.go]: oh I assumed expected error is an Error but it's a pattern

**Review by gssbzn — APPROVED** [review]

File: mongodbatlas/fw_resource_mongodbatlas_alert_configuration_test.go [hunk 2]
```diff
@@ -312,6 +313,33 @@ func TestAccConfigRSAlertConfiguration_importBasic(t *testing.T) {
...
+		Steps: []resource.TestStep{
+			{
+				Config: testAccMongoDBAtlasAlertConfigurationConfig(orgID, projectName, true),
+			},
+			{
+				ResourceName:            resourceName,
+				ImportStateIdFunc:       testAccCheckMongoDBAtlasAlertConfigurationImportStateIncorrectIDFunc(resourceName),
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"project_id"},
```
Comment (AgustinBettati) [inline mongodbatlas/fw_resource_mongodbatlas_alert_configuration_test.go]: nit: this line is redundant given that an error will be expected

File: mongodbatlas/fw_resource_mongodbatlas_alert_configuration_test.go [hunk 3]
```diff
@@ -312,6 +313,33 @@ func TestAccConfigRSAlertConfiguration_importBasic(t *testing.T) {
...
+		PreCheck:                 func() { testAccPreCheckBasic(t) },
+		ProtoV6ProviderFactories: testAccProviderV6Factories,
+		CheckDestroy:             testAccCheckMongoDBAtlasAlertConfigurationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccMongoDBAtlasAlertConfigurationConfig(orgID, projectName, true),
+			},
+			{
+				ResourceName:            resourceName,
+				ImportStateIdFunc:       testAccCheckMongoDBAtlasAlertConfigurationImportStateIncorrectIDFunc(resourceName),
```
Comment (AgustinBettati) [inline mongodbatlas/fw_resource_mongodbatlas_alert_configuration_test.go]: could we use ImportStateId directly here? (defining a fixed value with the incorrect format)

File: mongodbatlas/fw_resource_mongodbatlas_project_ip_access_list_test.go [hunk 4]
```diff
@@ -220,6 +221,32 @@ func TestAccProjectRSProjectIPAccessList_importBasic(t *testing.T) {
...
+		PreCheck:                 func() { testAccPreCheckBasic(t) },
+		ProtoV6ProviderFactories: testAccProviderV6Factories,
+		CheckDestroy:             testAccCheckMongoDBAtlasProjectIPAccessListDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccMongoDBAtlasProjectIPAccessListConfigSettingIPAddress(orgID, projectName, ipAddress, comment),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportStateIdFunc: testAccCheckMongoDBAtlasProjectIPAccessListImportStateIncorrectIDFunc(resourceName),
```
Comment (AgustinBettati) [inline mongodbatlas/fw_resource_mongodbatlas_project_ip_access_list_test.go]: same comment related to using ImportStateId

File: mongodbatlas/fw_resource_mongodbatlas_alert_configuration_test.go (hunk 2 above)
Comment (lantoli) [reply mongodbatlas/fw_resource_mongodbatlas_alert_configuration_test.go]: thx, changed in a new commit

File: mongodbatlas/fw_resource_mongodbatlas_alert_configuration_test.go (hunk 3 above)
Comment (lantoli) [reply mongodbatlas/fw_resource_mongodbatlas_alert_configuration_test.go]: thx, changed in a new commit

File: mongodbatlas/fw_resource_mongodbatlas_project_ip_access_list_test.go (hunk 4 above)
Comment (lantoli) [reply mongodbatlas/fw_resource_mongodbatlas_project_ip_access_list_test.go]: thx, changed in a new commit

**Review by AgustinBettati — APPROVED** [review]
LGTM, thanks for addressing comments

**Pull request merged**